service/src/main/java/io/camunda/service/ManagementServices.java

```diff
@@ -9,15 +9,16 @@
 
 import io.camunda.service.license.CamundaLicense;
 
-public final class ManagementService {
+public final class ManagementServices {
 
-  private final CamundaLicense camundaLicense;
+  private CamundaLicense camundaLicense;
 
-  public ManagementService() {
-    camundaLicense = new CamundaLicense();
+  public ManagementServices withLicense(final CamundaLicense license) {
+    camundaLicense = license;
+    return this;
   }
 
   public boolean isCamundaLicenseValid() {
-    return camundaLicense.isValid();
+    return camundaLicense != null && camundaLicense.isValid();
```

Comment:
❓ If it is null is it really valid? Do we maybe need an enum here to indicate DEV and prod? 🤔 And default to dev? 🤔

Reply:
I removed this null check completely, the license itself won't ever be `null` with how `managementServices` is being created

Reply:
I also started a conversation around dev/prod licenses, there seems to be a discrepancy if they are needed or not

https://camunda.slack.com/archives/C06V2SR6U9G/p1722973721151959

Reply:
From the linked slack conversation, the distinction between dev/prod licenses does not matter. The keys we will be issuing will always be production liceneses